ui/src/app/shared/components/dropdown.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import { DataLoader, Select, SelectOption } from 'argo-ui';
+
+interface Props {
+    load: Promise<(string | SelectOption)[]>;
+    onChange: (value: string) => void;
+}
+
+export class Dropdown extends React.Component<Props, { selected: string }> {
```

Comment:
Can I get you to change this into a stateless component? They are much more robust and I think we want to move away from class components.

https://reactjs.org/docs/hooks-intro.html

Reply:
>  functional component instead

Reply:
updated to functional component